### PR TITLE
Unify storage for sourceBundleHost/Port and debugHost

### LINKED ISSUE #5203: FastRefresh doesn't work with DebugHost
If an app specifies a custom DebugHost, we load the bundle from the server just fine, but call to 
`m_instance.Load()->callJSFunction("HMRClient", "setup", std::move(params));` uses GetSourceBundleHost and GetSourceBundlePort which doesn't respect the DebugHost setting.

We should either only support one of these mechanisms of specifying the host.  And/or make sure we are able to use the DebugHost setting for HMRClient.

### PR DESCRIPTION
**Pull request by acoates-ms**
Fixes #5203 

We have two settings on the instance settings.  We have DebugHost which is a string of format <host>:<port> which specifies the location of the packager.  We also have SourceBundleHost and SourceBundlePort, which is essentially the same thing, split out as two properties.

Internally we are inconsistent on which of these properties we use, which leads to issues like #5203, where some features dont work properly unless you set both of these properties to the same thing.   This change modifies the settings object so that setting DebugHost stores the data in the SourceBundleHost and SourceBundlePort property.  Thus preventing the settings from getting out of sync.  I also went through and ensured that we are using a shared source for the default values, as some of those values were repeated in a couple of places within the code.

I think long term we should remove DebugHost as a setting, since its a duplicate of the other settings. -- but for now we can keep both around.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5344)

### CODE REVIEW

**Comment by asklar** [comment]
CC @nasadigital

File: vnext/ReactWindowsCore/DevServerHelper.h [hunk 1]
```diff
@@ -15,45 +15,68 @@ class DevServerHelper {
...
+  }
+
+  static std::string get_PackagerConnectionUrl(const std::string &sourceBundleHost, const uint16_t sourceBundlePort) {
+    return string_format(PackagerConnectionUrlFormat, GetDeviceLocalHost(sourceBundleHost, sourceBundlePort).c_str());
+  }
+
+  static std::string get_PackagerOpenStackFrameUrl(
+      const std::string &sourceBundleHost,
+      const uint16_t sourceBundlePort) {
+    return string_format(PackagerConnectionUrlFormat, GetDeviceLocalHost(sourceBundleHost, sourceBundlePort).c_str());
```
Comment (TheSavior) [inline vnext/ReactWindowsCore/DevServerHelper.h:56]: Should this be PackagerOpenStackFrameUrlFormat?

Comment (acoates-ms) [reply vnext/ReactWindowsCore/DevServerHelper.h]: good catch

File: vnext/Microsoft.ReactNative/ReactInstanceSettings.h [hunk 2]
```diff
@@ -191,14 +191,6 @@ inline void ReactInstanceSettings::ByteCodeFileUri(hstring const &value) noexcep
   m_byteCodeFileUri = value;
 }
 
-inline hstring ReactInstanceSettings::DebugHost() noexcept {
-  return m_debugHost;
```
Comment (dannyvv) [inline vnext/Microsoft.ReactNative/ReactInstanceSettings.h:195]: >m_debugHost [](start = 9, length = 11)

Should we remove the backing field and initializer as well?

File: vnext/Microsoft.ReactNative/ReactHost/React.h [hunk 3]
```diff
@@ -108,7 +108,7 @@ struct ReactDevOptions {
   //! {EXTENSION} = ".bundle"
   //! Specify a value for a component, or leave empty to use the default.
   std::string SourceBundleHost; // Host domain (without port) for the bundler server. Default: "localhost".
-  std::string SourceBundlePort; // Host port for the bundler server. Default: "8081".
+  uint16_t SourceBundlePort{0}; // Host port for the bundler server. Default: "8081".
```
Comment (dannyvv) [inline vnext/Microsoft.ReactNative/ReactHost/React.h:111]: >0 [](start = 28, length = 1)

comment says 8081 as the default, shouldn't we initialzie with 8081 then?

Comment (dannyvv) [reply vnext/Microsoft.ReactNative/ReactHost/React.h:111]: -or- facebook::react::DevServerHelper::DefaultPackagerPort

---
In reply to: [445894069](https://github.com/microsoft/react-native-windows/pull/5344#discussion_r445894069) [](ancestors = 445894069)

File: vnext/Shared/OInstance.cpp [hunk 4]
```diff
@@ -497,7 +497,8 @@ void InstanceImpl::loadBundleInternal(std::string &&jsBundleRelativePath, bool s
       // First attempt to get download the Js locally, to catch any bundling
       // errors before attempting to load the actual script.
       auto jsBundleString = m_devManager->GetJavaScriptFromServer(
-          m_devSettings->debugHost,
```
Comment (vmoroz) [inline vnext/Shared/OInstance.cpp:500]: >m_devSettings->debugHost, [](start = 10, length = 25)

Do we need to do a corresponding change in Office code?

**Pull request merged**